### DISCUSSION TITLE
[RFC] Table support horizontal virtualization

### DISCUSSION THREAD

**Discussion by linxianxi** [ant-design/ant-design#48656] [RFCs]
## Summary
Table 支持水平虚拟滚动，默认不开启，否则可能会造成 breaking change。（比如行编辑，滚动后卸载了，保存时 Form onFinish 收集不到）

## Motivation
虽然列太多设计不合理，但是有些人确实需要。

## API

先支持 body 的虚拟，header 的先不支持（Virtual Table 还没有口子来修改 header 部分），header 就一行影响也不大

virtual 为 true 时，不开启水平虚拟滚动。

| prop | type |
| --- | --- |
|  virtual  | boolean \| { x? boolean } |

**Comment by afc163** [comment]
把 y 也支持了呗。

**Reply by linxianxi** [reply]
这个应该挺麻烦。开启x 关闭 y，就变成不使用 rc-virtual-list 了，变成普通 Table 了。这就是要在普通 Table 中再写一套不同的横向虚拟逻辑。唯一的好处应该就是集成行拖拽比较方便，目前的虚拟行拖拽滚动很麻烦且有一点点问题。
